client/alloc_runner.go

```diff
@@ -452,11 +477,22 @@ OUTER:
 	// Final state sync
 	r.syncStatus()
 
+	// Stop watching the shared allocation directory
+	r.ctx.AllocDir.StopDiskWatcher()
+
 	// Block until we should destroy the state of the alloc
 	r.handleDestroy()
 	r.logger.Printf("[DEBUG] client: terminating runner for alloc '%s'", r.alloc.ID)
 }
 
+// checkResources monitors and enforces alloc resource usage
+func (r *AllocRunner) checkResources() (bool, string) {
```

Comment:
Just have this return a *TaskEvent and the call site can check if it is nil. I would put a comment that this returns an appropriate task event describing why the allocation had to be killed.